### PR TITLE
Replace usages of deprecated pointer functions

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
This PR replaces the usages of deprecated pointer functions.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other dependency
Utility functions `QuantityPtr`,`ProtocolPtr`,`TimePtr` and `TimePtrDeref`, `extensionsv1alpha1.UnitCommandPtr` and `ValueExists` are dropped. Use `k8s.io/utils/ptr.To`, `k8s.io/utils/ptr.Deref` and `slices.Contains` instead.
```

### CODE REVIEW

**Comment by oliver-goetz** [comment]
Awesome, thanks for cleaning this up. There is no single `k8s.io/utils/pointer` left 🚀 

I found two additional custom pointer function you could delete too:
- https://github.com/shafeeqes/gardener/blob/aa57f82c90831e30d797571dd9634a78f985d39c/pkg/utils/kubernetes/highavailability_test.go#L90-L92
- https://github.com/shafeeqes/gardener/blob/aa57f82c90831e30d797571dd9634a78f985d39c/pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go#L141-L144

**Comment by shafeeqes** [comment]
> I found two additional custom pointer function you could delete too:
> 
> * https://github.com/shafeeqes/gardener/blob/aa57f82c90831e30d797571dd9634a78f985d39c/pkg/utils/kubernetes/highavailability_test.go#L90-L92

I considered this, but then we have to use `ptr.To(gardencorev1beta1.FailureToleranceType(<v>))` everywhere, which isn't any better. But for consistency reasons, I can do it.

> * https://github.com/shafeeqes/gardener/blob/aa57f82c90831e30d797571dd9634a78f985d39c/pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go#L141-L144

I considered this as well, but I saw that the extensions were using this and maybe let's keep it. I removed it and added to the release note.

**Comment by oliver-goetz** [comment]
Thanks 😄 
/lgtm

File: pkg/apis/operator/v1alpha1/validation/validation_test.go [hunk 1]
```diff
@@ -1204,24 +1204,24 @@ var _ = Describe("Validation Tests", func() {
...
-								Verbosity: pointer.Int32(3),
+								Verbosity: ptr.To(int32(3)),
 							}, BeEmpty()),
 							Entry("valid (httpAccessVerbosity>0)", &gardencorev1beta1.APIServerLogging{
-								HTTPAccessVerbosity: pointer.Int32(3),
+								HTTPAccessVerbosity: ptr.To(int32(3)),
 							}, BeEmpty()),
 							Entry("invalid (verbosity<0)", &gardencorev1beta1.APIServerLogging{
-								Verbosity: pointer.Int32(negativeSize),
+								Verbosity: ptr.To(int32(negativeSize)),
```
Comment (ary1992) [inline pkg/apis/operator/v1alpha1/validation/validation_test.go]:
```suggestion
								Verbosity: ptr.To(negativeSize),
```
`negativeSize` is of type `int32`.Similar changes for other occurrence of `negativeSize`.

**Review by rfranzke — COMMENTED** [review]
/approve

Thanks @shafeeqes 🙇🏻

**Review by ary1992 — COMMENTED** [review]
Thanks for the PR!
One minor change.

File: pkg/apis/operator/v1alpha1/validation/validation_test.go (hunk 1 above)
Comment (ary1992) [reply pkg/apis/operator/v1alpha1/validation/validation_test.go]: /hold

**Pull request merged**